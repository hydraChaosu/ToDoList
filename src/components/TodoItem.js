import React from "react"
import styled, { css } from "styled-components"

const Button = styled.button`
  padding: 15px 10px;
  min-width: 100px;
  margin: 5px;
  transition: 0.3s all;
  :hover {
    box-shadow: 0 0 10px yellow;
  }
  ${props =>
    props.com &&
    css`
      background: green;
      color: white;
    `}
  ${props =>
    props.rem &&
    css`
      background: red;
      color: white;
    `}
`
const List = styled.li`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  padding: 0 5px;
  text-decoration: none;
  list-style: none;
  @media (min-width: 768px) {
    padding: 0;
  }
`

const Content = styled.p`
  flex-grow: 1;
`

const TodoItem = ({ content, completed, complete, remove, id }) => {
  return (
    <List>
      <Content>{content}</Content>
      <Button com onClick={() => complete(id)}>
        {completed ? "Uncomplete" : "Complete"}
      </Button>
      <Button rem onClick={() => remove(id)}>
        Remove
      </Button>
    </List>
  )
}

export default TodoItem
