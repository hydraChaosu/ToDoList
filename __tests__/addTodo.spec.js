import React from "react"
import renderer from "react-test-renderer"

import ToDo from "../src/pages/index"
import TodoItem from "../src/components/TodoItem"
describe("ToDo", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<TodoItem id={20} key={20} content={"roman"} completed={false} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
