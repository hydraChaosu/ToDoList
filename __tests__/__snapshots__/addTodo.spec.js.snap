// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDo renders correctly 1`] = `
<li>
  <p>
    roman
  </p>
  <button
    onClick={[Function]}
  >
    Complete
  </button>
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</li>
`;
